### DISCUSSION TITLE
migration Version20200817133132 results in error post composer update

### DISCUSSION THREAD

**Discussion by neha04** [pimcore/pimcore#7983] [General]
While updating from pimcore 6.7.3 to 6.8.7 getting following issue

![Screenshot from 2021-01-25 18-08-00](https://user-images.githubusercontent.com/21308144/105709568-e71bcc80-5f3b-11eb-80e6-d50e71bd10a4.png)

**Comment by NiklasBr** [comment]
That migration has [caused issues before](/pimcore/pimcore/search?q=20200817133132&type=issues), but but should have been fixed by #7258

**Comment by neha04** [comment]
I have  not experienced this issue before. I dont know the reason but some of the relations table already have id column and some do not have.
Just changed the migrations file a little bit to resolve my issue

```
                if (!$schema->getTable($relationTable)->hasColumn('id')) {
                    // added this condition inside the loop
                    if ($schema->getTable($relationTable)->hasPrimaryKey()) {
                        $this->addSql('ALTER TABLE `' . $relationTable . '` DROP PRIMARY KEY;');
                    }
                    $this->addSql('ALTER TABLE `' . $relationTable . '` ADD COLUMN `id` BIGINT(20) NOT NULL PRIMARY KEY AUTO_INCREMENT FIRST');
                }
```
